### PR TITLE
Tweak: Add units to `container_width` and `widgets_gap` site settings [ED-9783]

### PR DESCRIPTION
**Pull request by rodolphebertozzo**

### CODE REVIEW

File: core/kits/documents/tabs/settings-layout.php [hunk 1]
```diff
@@ -52,9 +52,11 @@ protected function register_tab_controls() {
 		$this->add_responsive_control(
 			'container_width',
 			[
-				'label' => esc_html__( 'Content Width', 'elementor' ) . ' (px)',
+				'label' => esc_html__( 'Content Width', 'elementor' ),
 				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', '%', 'vw', 'custom' ],
 				'default' => [
+					'unit' => 'px',
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:59]: Not required, the default unit is always 'px'. It is defined in `Control_Base_Units()` class.

File: core/kits/documents/tabs/settings-layout.php [hunk 2]
```diff
@@ -105,21 +117,32 @@ protected function register_tab_controls() {
 		$this->add_control(
 			'space_between_widgets',
 			[
-				'label' => $widgets_space_label . ' (px)',
+				'label' => $widgets_space_label,
 				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', '%', 'vw', 'custom' ],
 				'default' => [
+					'unit' => 'px',
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:124]: Same here, this is not required.

File: core/kits/documents/tabs/settings-layout.php [hunk 3]
```diff
@@ -69,11 +71,21 @@ protected function register_tab_controls() {
...
+					'vw' => [
+						'min' => 50,
+						'max' => 100,
+						'step' => 1,
+					],
+					'description' => esc_html__( 'Sets the default width of the content area (Default: 1140px)', 'elementor' ),
+					'selectors' => [
+						'.elementor-section.elementor-section-boxed > .elementor-container' => 'max-width: {{SIZE}}{{UNIT}}',
+						'.e-con' => '--container-max-width: {{SIZE}}{{UNIT}}',
+					],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:88]: You moved the `description` and the `selectors` inside the `range`.

**Review by rami-elementor — COMMENTED** [review]
See the comments below.

File: core/kits/documents/tabs/settings-layout.php (hunk 1 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php (hunk 2 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php (hunk 3 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php [hunk 4]
```diff
@@ -52,8 +52,9 @@ protected function register_tab_controls() {
 		$this->add_responsive_control(
 			'container_width',
 			[
-				'label' => esc_html__( 'Content Width', 'elementor' ) . ' (px)',
+				'label' => esc_html__( 'Content Width', 'elementor' ),
 				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', '%', 'vw', 'custom' ],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:57]: Add `em` and `rem` units.

File: core/kits/documents/tabs/settings-layout.php [hunk 5]
```diff
@@ -69,8 +70,18 @@ protected function register_tab_controls() {
...
+					'%' => [
+						'min' => 50,
+						'max' => 100,
+						'step' => 1,
+					],
+					'vw' => [
+						'min' => 50,
+						'max' => 100,
+						'step' => 1,
+					],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:82]: Remove the range values. Elementor will inherit this from `Control_Base_Units()` class.

File: core/kits/documents/tabs/settings-layout.php [hunk 6]
```diff
@@ -105,8 +116,9 @@ protected function register_tab_controls() {
 		$this->add_control(
 			'space_between_widgets',
 			[
-				'label' => $widgets_space_label . ' (px)',
+				'label' => $widgets_space_label,
 				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 'px', '%', 'vw', 'custom' ],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:121]: Add `em` and `rem` units.

File: core/kits/documents/tabs/settings-layout.php [hunk 7]
```diff
@@ -115,11 +127,20 @@ protected function register_tab_controls() {
...
 						'max' => 40,
 					],
+					'%' => [
+						'min' => 0,
+						'max' => 20,
+					],
+					'vw' => [
+						'min' => 0,
+						'max' => 40,
+					],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:137]: Remove.

File: core/kits/documents/tabs/settings-layout.php [hunk 8]
```diff
@@ -115,11 +127,20 @@ protected function register_tab_controls() {
...
 				],
 				'placeholder' => [
-					'size' => '20',
+					'size' => 20,
+					'unit' => 'px',
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:141]: revert

**Review by rami-elementor — COMMENTED** [review]
The task was to add units to **Global Settings**. Focus on `core/document-types/page-base.php` file, try updating only the `size_units` argument. Add `em`, `rem` and `vw` units.

File: core/kits/documents/tabs/settings-layout.php (hunk 8 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php:124]: Done.

File: core/kits/documents/tabs/settings-layout.php (hunk 7 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php (hunk 6 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php (hunk 5 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php (hunk 4 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done.

File: core/kits/documents/tabs/settings-layout.php [hunk 9]
```diff
@@ -105,21 +106,20 @@ protected function register_tab_controls() {
 		$this->add_control(
 			'space_between_widgets',
 			[
-				'label' => $widgets_space_label . ' (px)',
+				'label' => $widgets_space_label,
 				'type' => Controls_Manager::SLIDER,
-				'default' => [
-					'size' => 20,
-				],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:112]: You removed the default values... Why?

File: core/kits/documents/tabs/settings-layout.php [hunk 10]
```diff
@@ -105,21 +106,20 @@ protected function register_tab_controls() {
...
 				'range' => [
 					'px' => [
 						'min' => 0,
 						'max' => 40,
 					],
 				],
 				'placeholder' => [
-					'size' => '20',
+					'size' => 20,
+					'unit' => 'px',
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:120]: Remove the units. It's not required. Focus mainly on `size_units` in this PR.

**Review by rami-elementor — CHANGES_REQUESTED** [review]

File: core/kits/documents/tabs/settings-layout.php (hunk 9 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php:112]: Error. My bad

File: core/kits/documents/tabs/settings-layout.php (hunk 10 above)
Comment (rodolphebertozzo) [reply core/kits/documents/tabs/settings-layout.php]: Done. Normally everything is good, sorry for any confusion, I wanted to make something really complete, without knowing more about the architecture of the code regarding the units.

**Review by rami-elementor — DISMISSED** [review]

File: core/kits/documents/tabs/settings-layout.php [hunk 11]
```diff
@@ -105,21 +106,22 @@ protected function register_tab_controls() {
 		$this->add_control(
 			'space_between_widgets',
 			[
-				'label' => $widgets_space_label . ' (px)',
+				'label' => $widgets_space_label,
 				'type' => Controls_Manager::SLIDER,
 				'default' => [
 					'size' => 20,
 				],
+				'size_units' => [ 'px', 'em', '%', 'rem', 'vw', 'custom' ],
```
Comment (rami-elementor) [inline core/kits/documents/tabs/settings-layout.php:114]:
```suggestion
				'size_units' => [ 'px', 'em', 'rem', 'custom' ],
```

**Review by arielk — APPROVED** [review]

**Pull request merged**